ls1intum/Artemis
# [Feature] Implement new assessment type for programming exercises

### Checklist
- [ ] Server: I added multiple integration tests (Spring) related to the features. **Will come on follow up PR**
- [ ] Client: I added multiple integration tests (Jest) related to the features. **Will come on follow up PR**

### Motivation and Context


Sometimes automatic tests already have a good coverage and therefore assessment quality. So a tutor doe not have to look over every submission. But in some cases the tests are not sufficient, which makes a manual assessment necessary.
This PR adds a new AssessmentType called "Complaint based, which will normally grade students based on the tests, but allows them to complain and get a manual assessment.

### Description

This PR adds the option for programming exercises that the students can complain after the deadline without a manual assessment for that exercise. This option can be enabled in the timeline of the programming exercise when it is an exam exercise or when complains are enabled in the course.

### Steps for Testing

1. Create exams and exercises and make sure that the students can complain after a manual assessment as always
2. Create exams and exercises and make sure that the students **can't** complain after an automatic assessment
2. Create programming exercises in the course and an exam and select the complain based assessment (the icon under Assessment type in the timeline should show gears)
3. Check that the new assessment type is only available in exams and courses where complains are enabled
4. Check that the students can complain after the deadline in these exercises without getting a manual assessment.
5. Check that these complains can be assessed by a tutor and can lead to a score change when accepted. (The complains are currently only displayed in the list of all complains and not on the assessment dashboard)

### Test Coverage



Will be added in followup PR

### Screenshots
